contentcuration/contentcuration/static/js/edit_channel/import/views/ImportListItem.vue

```diff
@@ -0,0 +1,269 @@
+          <transition-group name="fade">
+            <ImportListItem
+              ref="children"
+              v-for="file in subFiles"
+              :key="file.id"
+              :node="file"
+              :isFolder="file.kind ==='topic'"
+              :isChannel="false"
+              :parentIsChecked="isChecked"
+            />
+          </transition-group>
+        </ul>
+      </div>
+  </li>
+
+</template>
+
+
+<script>
+
+const _ = require('underscore');
+const RequiredBoolean = { type: Boolean, required: true };
+const stringHelper = require('../../utils/string_helper');
+const { fetchContentNodesById, getIconClassForKind } = require('../util');
+const { mapActions } = require('vuex');
+const { pluralize } = require('./filters');
+
+module.exports = {
+  name: 'ImportListItem',
+  props: {
+    isChannel: RequiredBoolean,
+    isFolder: RequiredBoolean,
+    isRoot: {
+      type: Boolean,
+      default: false,
+    },
+    parentIsChecked: {
+      type: Boolean,
+      default: false,
+    },
+    // node :: { title: String, children: Array }
+    node: {
+      type: Object,
+      required: true,
+    }
+  },
+  data() {
+    return {
+      isLoading: false,
+      isChecked: false,
+      subFiles: [],
+      isExpanded: false,
+      childrenAreLoaded: false,
+    }
+  },
+  mounted() {
+    this.isChecked = this.parentIsChecked;
+  },
+  watch: {
+    parentIsChecked(newVal) {
+      // if parent is suddenly checked, remove this node from the list
+      if (newVal && this.isChecked) {
+        this.removeItemFromImportList(this.node.id);
+      }
+      this.isChecked = newVal;
+    }
+  },
+  computed: {
+    togglerClass() {
+      return {
+        'glyphicon': true,
+        'glyphicon-triangle-right': !this.isExpanded,
+        'glyphicon-triangle-bottom': this.isExpanded,
+      };
+    },
+    resourceCount() {
+      return this.node.metadata.resource_count;
+    },
+    iconClass() {
+      return {
+        glyphicon: true,
+        [getIconClassForKind(this.node.kind)]: true,
+      };
+    },
+    importListItemClass() {
+      return {
+        disabled: this.isDisabled || this.parentIsChecked,
+      }
+    },
+    resourceSize() {
+      return stringHelper.format_size(this.node.metadata.resource_size);
+    },
+    hasChildren() {
+      return this.node.children.length > 0;
+    }
+  },
+  methods: {
+    ...mapActions('import', [
+      'addItemToImportList',
+      'removeItemFromImportList',
+    ]),
+    fetchChildData() {
+      // If children are loaded once, then do nothing
+      if (this.childrenAreLoaded) return;
+      this.isLoading = true;
+      return fetchContentNodesById(this.node.children)
+      .then((childData) => {
+        this.isLoading = false;
+        this.childrenAreLoaded = true;
+        this.subFiles = childData;
+      });
+    },
+    handleClickLabel() {
+      const isToggleable = this.isChannel || this.hasChildren;
+      if (isToggleable) {
```

Comment:
Not sure if this was intentional, but opening a topic, opening a subtopic, closing the topic, reopening the topic, and reopening the subtopic reloads the content. Might be good to keep the elements loaded but hidden so that the content doesn't need to get reloaded on each toggle (especially for large topics like the Khan Academy topics). Also, items that are checked don't get rendered as checked when they are reloaded
![import](https://user-images.githubusercontent.com/7447496/28042398-e939fa62-6581-11e7-8238-29ea39b31687.gif)